cmd/snap-bootstrap/inputwatch/inputwatch.go

```diff
@@ -0,0 +1,99 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package inputwatch
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/snapcore/snapd/logger"
+)
+
+type inputProvider interface {
+	FindMatchingDevices(filter inputEventFilter) ([]inputDevice, error)
+}
+
+type keyEvent struct {
+	Dev inputDevice
+	Err error
+}
+
+type inputDevice interface {
+	WaitForTrigger(chan keyEvent)
+	String() string
+}
+
+type inputEventFilter struct {
+	Key string
+}
+
+var (
+	// input mechanism
+	input inputProvider
+
+	// wait for '1' to be pressed
+	triggerFilter = inputEventFilter{Key: "KEY_1"}
```

Comment:
if the key is hardcoded in the package, the package name feels a bit wrong,  triggerwatch ?

Reply:
s/input/trigger/ and package rename is done now